### DISCUSSION TITLE
Wordpress - Find all images less than 100px width and Delete

### DISCUSSION THREAD

**Discussion by taylorcurry** [ImageMagick/ImageMagick#6232] [General]
Hello - 

I have a large wordpress server with over 800k media files. Could ImageMagick help me find all images under 100px wide and delete them? 

If so what would be the command or best way to do this?

Thank you

Taylor

**Comment by fmw42** [comment]
Sorry, I do not use Windows. But I think you would need to write a small script FOR loop over your directories and then then files within the directories. The loop would then call Imagemagick command to get the width and test it against your threshold. If too large, then you would call your OS to delete the file or move it to some other backup directory.

On Linux or Mac OSX, I would search each directory as


```
cd to directory
for img in *; do
test=$( magick "$img" -format "%[fx:(B>100000)?1:0]\n" info: )
[ $test -eq 1 ] && rm "$img"
done
```

In the above I test if B is greater than 100kb and if so, I use the OS to remove the file. Note that %B is the file size in bytes, so I convert 100kb to 100000

One can add a FOR loop over each directory as well in the above.

**Comment by snibgo** [comment]
I don't know if the server is Windows or Linux or whatever. But the answer is the same: IM has no facility to delete files. "`magick identify -format %w infile.ext`" will tell you width of the image in a file, and you can use that in a script to do whatever you want.

**Comment by GeeMack** [comment]
There may be better tools to do this task, but with ImageMagick v7 you can generate a list of images smaller than 100 pixels wide with a Windows command like this run in a "for" loop through your directory of images...

    magick input.jpg -set filename:0 "%[f]\n" ( +clone -set filename:0 "" ) -delete "%[fx:w<100]" -print "%[filename:0]" null:

That reads the input image, sets a filename variable to its original name, makes a clone of it and sets that filename variable to just empty quotes "". Then it deletes one or the other of those in-memory images according to width test "%[fx:w<100]". This command finishes by printing the output of either the input file name if that input is narrower than 100 pixels, or if the image is 100 pixels or wider it prints nothing.

Some creative tweaking of the command to print the output to a list from a Windowss BAT script might look like this...

    @echo off

    del list.txt 2>NUL

    for %%I in ( *.png *.jpg ) do (
       magick "%%I" -set filename:0 "%%I\n" ^( +clone -set filename:0 "" ^) ^
          -delete "%%[fx:w<100]" -print "%%[filename:0]" null: >> list.txt )

    exit /b

Finish that script by deleting all the files in the "list.txt" or manually delete them after you create the list.

These commands are in Windows syntax. I wouldn't know right off and haven't had time to explore how to build that "for" command on a *nix OS.

**Comment by snibgo** [comment]
Inventive as ever, GeeMack!

Your command...
```
del list.txt 2>NULL
```
... will create a file named `NULL`. I think you intended to send the output to the `NUL` device (only one `L`):
```
del list.txt 2>NUL
```

**Reply by GeeMack** [reply]
Fixed it. Thanks!

**Reply by taylorcurry** [reply]
What about something like this? 

`find . -type f -name '*.jpg' -exec identify -format '%w %h %i\n' '{}' \; | awk '$1<100 {print $3}'`

However when i attempt to run the command inside a folder it returns this : 

`find: ‘identify’: No such file or directory`

**Comment by snibgo** [comment]
> find: ‘identify’: No such file or directory

Does "`identify`" work from the command line?

What is your version of IM? On what platform?
